tests/integration/nm/ovs_test.py

```diff
@@ -260,3 +264,136 @@ def test_ovs_bridge_remoev_external_ids(self, disable_ovsdb_plugin):
+        "0.0.0.0/0 192.0.2.1",
+    )
+    cmdlib.exec_cmd(nmcli_ovs_interface, check=True)
+
+
+def _verify_ovs_activated(ovs_name):
+    ret, out, err = cmdlib.exec_cmd(
+        f"nmcli --field GENERAL.STATE device show {ovs_name}".split(),
+        check=True,
+    )
+    connected = "connected" in out
+    ret, out, err = cmdlib.exec_cmd(
+        f"nmcli --field IP4.ADDRESS device show {ovs_name}".split(),
+        check=True,
+    )
+    ipv4_configured = "192.0.2.2/24" in out
+    ret, out, err = cmdlib.exec_cmd(
+        f"nmcli --field IP4.ROUTE device show {ovs_name}".split(),
+        check=True,
+    )
+    route_configured = "0.0.0.0/0" in out
+    return connected and ipv4_configured and route_configured
+
+
+@pytest.fixture
+def ovs_bridge_first_and_ovs_interface_with_same_name_ipv4():
+    # The order on this function is important. The OVS bridge must be defined
+    # before the OVS interface.
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-port conn.interface br-ex master br-ex "
+        "con-name ovs-port-br-ex".split(),
+        check=True,
+    )
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-bridge con-name br-ex conn.interface "
+        "br-ex".split(),
+        check=True,
+    )
+    _nmcli_ovs_bridge_with_ipv4_dns()
+    # Wait a little bit for NM to activate above interfaces to do not hit race
+    # problems.
+    assert retry_till_true_or_timeout(
+        VERIFY_RETRY_TMO, _verify_ovs_activated, OVS_DUP_NAME
+    )
+    yield
+    cmdlib.exec_cmd(
+        "nmcli connection del ovs-port-br-ex br-ex ovs-if-br-ex".split(),
+        check=True,
+    )
+
+
+@pytest.fixture
+def ovs_interface_first_and_ovs_bridge_with_same_name_ipv4():
+    # The order on this function is important. The OVS interface must be
+    # defined before the OVS bridge.
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-port conn.interface br-ex master br-ex "
+        "con-name ovs-port-br-ex".split(),
+        check=True,
+    )
+    _nmcli_ovs_bridge_with_ipv4_dns()
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-bridge con-name br-ex conn.interface "
+        "br-ex".split(),
+        check=True,
+    )
+    # Wait a little bit for NM to activate above interfaces to do not hit race
+    # problems.
+    assert retry_till_true_or_timeout(
+        VERIFY_RETRY_TMO, _verify_ovs_activated, OVS_DUP_NAME
+    )
+    yield
+    cmdlib.exec_cmd(
+        "nmcli connection del ovs-port-br-ex br-ex ovs-if-br-ex".split(),
+        check=True,
+    )
+
+
+@pytest.mark.tier1
+def test_modify_state_with_ovs_dup_name_ovs_bridge_first_with_ipv4_dns(
+    ovs_bridge_first_and_ovs_interface_with_same_name_ipv4,
+):
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: ETH1,
+                Interface.TYPE: InterfaceType.ETHERNET,
+                Interface.STATE: InterfaceState.UP,
+            }
+        ]
+    }
+    libnmstate.apply(desired_state)
```

Comment:
Since we are supposed to work well on supporting this duplicate name use case, can you use `assert_state_match` on old OVS interfaces.

Reply:
`assert_state_match` does not support duplicate interfaces. It requires some changes, I am adding this to the issue #1557. I hope this could be merged without it for now.